packages/target-decisioning-engine/src/artifactProvider.js

```diff
@@ -95,6 +96,26 @@ async function ArtifactProvider(config) {
             lastResponse = res.clone();
             lastEtag = etag;
           }
+          const latitude = res.headers.get("X-GEO-Latitude");
+          if (latitude != null && !isUndefined(latitude)) {
+            geoContext.latitude = latitude;
+          }
+          const longitude = res.headers.get("X-GEO-Longitude");
+          if (longitude != null && !isUndefined(longitude)) {
+            geoContext.longitude = longitude;
+          }
+          const countryCode = res.headers.get("X-GEO-Country-Code");
+          if (countryCode != null && !isUndefined(countryCode)) {
+            geoContext.countryCode = countryCode;
+          }
+          const stateCode = res.headers.get("X-GEO-Region-Code");
+          if (stateCode != null && !isUndefined(stateCode)) {
+            geoContext.stateCode = stateCode;
+          }
+          const city = res.headers.get("X-GEO-City");
+          if (city != null && !isUndefined(city)) {
+            geoContext.city = city;
+          }
```

Comment:
What do you think about moving this logic to a new file called `geoProvider` with a method on it that takes the response and returns geoContext?

Reply:
Sure, moved it to a new file.